JSDemos/Demos/Toolbar/Multiline/React/App.js

```diff
@@ -0,0 +1,358 @@
+import React from 'react';
+import Toolbar, { Item } from 'devextreme-react/toolbar';
+import { Button } from 'devextreme-react/button';
+import { ButtonGroup } from 'devextreme-react/button-group';
+import RadioGroup from 'devextreme-react/radio-group';
+import Resizable from 'devextreme-react/resizable';
+import CheckBox from 'devextreme-react/check-box';
+import DateBox from 'devextreme-react/date-box';
+import DropDownButton from 'devextreme-react/drop-down-button';
+import SelectBox from 'devextreme-react/select-box';
+import notify from 'devextreme/ui/notify';
+import 'devextreme/ui/select_box';
+import {
+  colors,
+  fontSizes,
+  lineHeights,
+  fonts,
+  fontStyles,
+  textAligns,
+  listTypes,
+} from './data.js';
+
+function getTextAlignItems(isExtended) {
+  return !isExtended
+    ? textAligns.map(({ icon, alignment, hint }) => ({ icon, alignment, hint }))
+    : textAligns;
+}
+
+const dropDownOptions = { width: 'auto' };
+const priorities = ['Multiline mode', 'Singleline mode'];
+
+const attachButtonOptions = {
+  icon: 'attach',
+  text: 'Attach',
+  onClick: () => {
+    notify('Attach button has been clicked!');
+  },
+};
+
+const addButtonOptions = {
+  icon: 'add',
+  text: 'Add',
+  onClick: () => {
+    notify('Add button has been clicked!');
+  },
+};
+
+const removeButtonOptions = {
+  icon: 'trash',
+  text: 'Remove',
+  onClick: () => {
+    notify('Remove button has been clicked!');
+  },
+};
+
+const aboutButtonOptions = {
+  icon: 'help',
+  text: 'About',
+  onClick: () => {
+    notify('About button has been clicked!');
+  },
+};
+
+function App() {
+  const colorPicker = React.useRef(null);
+
+  const [lineHeight, setLineHeight] = React.useState(lineHeights[1].lineHeight);
+  const [fontSize, setFontSize] = React.useState(fontSizes[2].size);
+  const [textAlign, setTextAlign] = React.useState([textAligns[0].alignment]);
+  const [priority, setPriority] = React.useState(priorities[0]);
+  const [multiline, setMultiline] = React.useState(true);
+  const [checkBoxValue, setCheckBox] = React.useState(false);
+
+  const onCheckBoxChangedHandler = React.useCallback(
+    ({ value }) => {
+      setCheckBox(value);
+      notify('Private option has been changed!');
+    },
+    [checkBoxValue, setCheckBox],
+  );
+
+  const onSelectionChangedHandler = React.useCallback(
+    (e) => {
+      setLineHeight(e.item.lineHeight);
+    },
+    [lineHeight, setLineHeight],
+  );
+
+  const onFontSizeChangedHandler = React.useCallback(
+    (e) => setFontSize(e.item.size),
+    [fontSize, setFontSize],
+  );
+
+  const onTextAlignChangedHandler = React.useCallback(
+    (e) => setTextAlign([e.item.alignment]),
+    [textAlign, setTextAlign],
+  );
+
+  const onChangeHandler = React.useCallback(({ value }) => {
+    setMultiline(value === priorities[0]);
+    setPriority(value);
+  }, []);
+
+  const onColorClick = React.useCallback((colorCode) => {
```

Comment:
It seems strange to me that `onColorClick` is memoized here.
I think, if we really want to show this palette,  it would be optimal to create separate component for this..

Reply:
Removed in https://github.com/DevExpress/devextreme-demos/pull/2004/commits/5ba43954aa5e53c1a0c2a8596281ef2bf941d78c